lib/rack/lti/middleware.rb

```diff
@@ -77,11 +77,13 @@ def valid_nonce?(nonce)
     end
 
     def valid_timestamp?(timestamp)
-      if @config.time_limit.nil?
-        true
-      else
-        (Time.now.to_i - @config.time_limit) <= timestamp
-      end
+      valid = true
+      now = Time.now.to_i
+
+      valid = (now - config.time_limit) <= timestamp if config.time_limit
+      valid = (now + config.future_time_limit) >= timestamp if valid && config.future_time_limit
```

Comment:
Is it possible for the first check on line 83 to set `valid` to false, only for this line to set it to true? I feel it would be safer to short-circuit return here instead of using the `valid` variable.

Reply:
@omarkhan thanks for eyes. It *should not* be possible for that scenario to occur because the conditional on line 84 includes the following:

`if valid && config.future_time_limit`

put another way, _"is the request still valid (not too far into the past from our check on line 83) and do we care to also check the future time as well if the user provided a value?"_

we could of course return false immediately as soon as any one check fails, and I'm happy to rework this another way if folks find the code here a bit unclear!

Reply:
Ah cool I missed the `if valid && ...` conditional. I still think short-circuit return would be clearer, but not a big deal.

Reply:
Will adjust!